src/Build/BackEnd/Shared/BuildRequestConfiguration.cs

```diff
@@ -170,15 +175,15 @@ internal BuildRequestConfiguration(int configId, BuildRequestData data, string d
             _explicitToolsVersionSpecified = data.ExplicitToolsVersionSpecified;
             _toolsVersion = ResolveToolsVersion(data, defaultToolsVersion);
             _globalProperties = data.GlobalPropertiesDictionary;
-            TargetNames = new List<string>(data.TargetNames);
+            RequestedTargets = new List<string>(data.TargetNames);
 
             // The following information only exists when the request is populated with an existing project.
             if (data.ProjectInstance != null)
             {
                 _project = data.ProjectInstance;
                 _projectInitialTargets = data.ProjectInstance.InitialTargets;
                 _projectDefaultTargets = data.ProjectInstance.DefaultTargets;
-
+                _projectTargets = data.ProjectInstance.Targets.Keys.ToHashSet();
```

Comment:
NIT: `.Targets.Keys.ToHashSet()` repeats three times in scope of this document. We can either create private method `TargetsOfProjectInstace(ProjectInstance pi)` or getter `ProjectInstance.TargetsList`.